src/processor/operator/persistent/node_batch_insert.cpp

```diff
@@ -97,23 +92,30 @@ void NodeBatchInsert::initLocalStateInternal(ResultSet* resultSet, ExecutionCont
     }
     // NOLINTEND(bugprone-unchecked-optional-access)
 
-    KU_ASSERT(state != nullptr);
-    for (auto i = 0u; i < nodeInfo->columnPositions.size(); ++i) {
-        auto pos = nodeInfo->columnPositions[i];
-        if (pos.isValid()) {
-            nodeLocalState->columnVectors.push_back(resultSet->getValueVector(pos).get());
+    for (auto i = 0u; i < nodeInfo->columnEvaluators.size(); ++i) {
```

Comment:
Can u separate the handling of default and non-default vectors into different for loop? that looks easier to understand to me.
The first loop: 1. evaluate non-default evaluators and set column vectors; 2. figure out `state`.
The second loop: 1. create new local vector for default columns, set state for them, and add them to column vectors.